src/wallet/rpcwallet.cpp

```diff
@@ -726,6 +742,113 @@ UniValue getbalance(const UniValue& params, bool fHelp)
     return ValueFromAmount(nBalance);
 }
 
+UniValue getbalancedetail(const UniValue& params, bool fHelp)
+{
+    if (fHelp || params.size() > 2)
+         throw runtime_error(
+                "getbalance ( minconf includeWatchonly )\n"
+                "\n"
+                "\nArguments:\n"
+                "1. minconf          (numeric, optional, default=1) Only include transactions confirmed at least this many times.\n"
+                "2. includeWatchonly (bool, optional, default=false) Also include balance in watchonly addresses (see 'importaddress')\n"
+                "\nResult:\n"
+                "detailed list       (JSON) A list of outputs similar to listtransactions that compose the entire balance.\n"
+                );
+
+    LOCK2(cs_main, pwalletMain->cs_wallet);
+
+    int nMinDepth = 1;
+    isminefilter filter = ISMINE_SPENDABLE;
+
+    if (params.size() > 0)
+    {
+        nMinDepth = params[0].get_int();
+        if (params.size() > 1)
+        {
+            if (params[1].get_bool()) filter = filter | ISMINE_WATCH_ONLY;
+        }
+    }
+
+    UniValue ret(UniValue::VOBJ);
+    UniValue items(UniValue::VARR);
+
+        // Calculate total balance a different way from GetBalance()
+        // (GetBalance() sums up all unspent TxOuts)
+        int64_t nBalance = 0;
+        int64_t totalFee = 0;
+
+        for (map<uint256, CWalletTx>::iterator it = pwalletMain->mapWallet.begin(); it != pwalletMain->mapWallet.end(); ++it)
+        {
+            const CWalletTx& wtx = it->second;
+            if (!wtx.IsTrusted())
+                continue;
+
+            int64_t allFee;
+            string strSentAccount;
+            list<COutputEntry> listReceived;
+            list<COutputEntry> listSent;
+
+            wtx.GetAmounts(listReceived, listSent, allFee, strSentAccount, filter);
+
+            if (wtx.GetDepthInMainChain() >= nMinDepth && wtx.GetBlocksToMaturity() == 0)
+            {
+                for (auto const& r : listReceived)
+                {
+                    CBitcoinAddress addr;
+                    addr.Set(r.destination);
+
+                    nBalance += r.amount;
+
+                    UniValue item(UniValue::VOBJ);
+
+                    item.pushKV("timestamp", (int64_t) wtx.nTime);
+                    item.pushKV("hash", wtx.GetHash().ToString());
```

Comment:
```suggestion
                    item.pushKV("txid", wtx.GetHash().ToString());
```

(for consistency with the other RPCs)

Reply:
May also be good to cache `wtx.GetHash().ToString()` in a local because it hashes the whole transaction for each `r` and `s`.